src/StreamJsonRpc/CorrelationManagerTracingStrategy.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+    using Microsoft;
+    using StreamJsonRpc.Protocol;
+
+    /// <summary>
+    /// Synchronizes activities as set by the <see cref="CorrelationManager"/> class over RPC.
+    /// </summary>
+    /// <seealso cref="ActivityTracingStrategy"/>
+    public class CorrelationManagerTracingStrategy : IActivityTracingStrategy
+    {
+        private static readonly AsyncLocal<string?> TraceStateAsyncLocal = new AsyncLocal<string?>();
+
+        /// <summary>
+        /// Gets or sets the contextual <c>tracestate</c> value.
+        /// </summary>
+        public static string? TraceState
+        {
+            get => TraceStateAsyncLocal.Value;
+            set => TraceStateAsyncLocal.Value = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="System.Diagnostics.TraceSource"/> that will receive the activity transfer, start and stop events .
+        /// </summary>
+        public TraceSource? TraceSource { get; set; }
+
+        /// <inheritdoc/>
+        public unsafe void ApplyOutboundActivity(JsonRpcRequest request)
+        {
+            Requires.NotNull(request, nameof(request));
+
+            if (Trace.CorrelationManager.ActivityId != Guid.Empty)
+            {
+                var traceparent = default(TraceParent);
+
+                FillRandomBytes(new Span<byte>(traceparent.ParentId, TraceParent.ParentIdByteCount));
+                CopyGuidToBuffer(Trace.CorrelationManager.ActivityId, new Span<byte>(traceparent.TraceId, TraceParent.TraceIdByteCount));
+
+                if (this.TraceSource is object && (this.TraceSource.Switch.Level & SourceLevels.ActivityTracing) == SourceLevels.ActivityTracing && this.TraceSource.Listeners.Count > 0)
```

Comment:
`x is object` is exactly equivalent in generated code to `x != null` except that `x is object` won't compile when `x` is a `struct` whereas `x != null` will compile and allow the misperception that the check is useful to continue. So the C# team's recommendation is to use `is object` instead of `!= null` in all instances. By adopting this pattern myself, I've caught several bugs in my own code and others'.
Similarly `x is null` is preferred over `x == null` for all the same reasons.

`HasFlag` is easier on the eyes, there's no doubt. But it *boxes* the enum. While that might seem small, in code that's hit for every message (like this will be) that cost adds up quickly. So quickly, in fact, that it was the no. 1 allocated object in a perf trace I just ran and had to [fix it](https://github.com/microsoft/vs-streamjsonrpc/commit/db7f844ad32ca5cc236252d715187b0c2ad5c356).